src/backend/api/Fusion.Resources.Api/Notifications/ContractAllocationNotificationHandler.cs

```diff
@@ -0,0 +1,122 @@
+                    .WithDescriptionMarkdown(NotificationDescription.DelegateAssigned(delegatedRole)));
+        }
+
+        public async Task Handle(DelegatedContractRoleRecertified notification, CancellationToken cancellationToken)
+        {
+            var delegatedRole = await mediator.Send(new GetContractDelegatedRole(notification.RoleId));
+
+            if (delegatedRole == null)
+                return;
+
+            await notificationClient.CreateNotificationAsync(n => n
+                    .WithRecipient(delegatedRole.Person.AzureUniqueId)
+                    .WithTitle($"Your delegated role in {delegatedRole.Contract.ContractNumber} - {delegatedRole.Project.Name} was recertified")
+                    .WithDescriptionMarkdown(NotificationDescription.DelegateRecertified(delegatedRole)));
+        }
+
+        public async Task Handle(CompanyRepUpdated notification, CancellationToken cancellationToken)
+        {
+            await CreateNotificationForPositionAsync(notification.PositionId, "You were allocated as Company Rep");
+        }
+
+        public async Task Handle(ContractRepUpdated notification, CancellationToken cancellationToken)
+        {
+            await CreateNotificationForPositionAsync(notification.PositionId, "You were allocated as Contract Rep");
+        }
+
+        public async Task Handle(ExternalCompanyRepUpdated notification, CancellationToken cancellationToken)
+        {
+            await CreateNotificationForPositionAsync(notification.PositionId, "You were allocated as External Company Rep");
+        }
+
+        public async Task Handle(ExternalContractRepUpdated notification, CancellationToken cancellationToken)
+        {
+            await CreateNotificationForPositionAsync(notification.PositionId, "You were allocated as External Contract Rep");
+        }
+
+        private async Task CreateNotificationForPositionAsync(Guid positionId, string title)
+        {
+            var position = await orgApiClient.GetPositionV2Async(positionId);
+            var instance = position?.GetActiveInstance();
+
+            if (position == null || instance?.AssignedPerson?.AzureUniqueId == null)
+                return;
+
+            await notificationClient.CreateNotificationAsync(n => n
+                .WithRecipient(instance.AssignedPerson.AzureUniqueId)
+                .WithTitle(title)
+                .WithDescriptionMarkdown(NotificationDescription.PositionAssigned(position, instance)));
+        }
+
+        private static class NotificationDescription
+        {
+            public static string DelegateRecertified(QueryDelegatedRole role) => new MarkdownDocument()
+                .Paragraph($"{MdToken.Bold(role.RecertifiedBy?.Name)} ({role.RecertifiedBy?.Mail}) recertified your {role.Type} role.")
+                .List(l => l
+                    .ListItem($"{MdToken.Bold("Project:")} {role.Project?.Name}")
```

Comment:
Again, this is just to avoid crashing the whole recertification request if for some reason the data would be corrupted. 
The role should always have a project in the case when a notification being sent. 
In case of an error, the text will just be empty.